File: packages/js-sdk/src/regex/detectors/ie/pps.ts
// Irish Personal Public Service (PPS) Number detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { iePpsChecksum } from '../../validators'

class IePpsDetector extends RegexDetector {
  entityType = EntityType.IE_PPS
  score = 0.85
  contextKeywords = [
    'pps',
    'ppsn',
    'personal public service',
    'revenue',
  ]
  contextRequired = true
  contextWindow = 50

  pattern = /\b\d{7}[A-Z]{1,2}\b/gi
  validator = iePpsChecksum
}

registerRegion('ie', IePpsDetector)

export { IePpsDetector }

File: packages/python-sdk/src/blindfold/regex/detectors/ie/pps.py
"""Irish Personal Public Service (PPS) Number detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import ie_pps_checksum


@register_region("ie")
class PpsDetector(RegexDetector):
    entity_type = EntityType.IE_PPS
    score = 0.85
    context_keywords = [
        "pps", "ppsn", "personal public service", "revenue",
    ]
    context_required = False
    context_window = 50

    pattern = re.compile(
        r"\b\d{7}[A-Z]{1,2}\b"
    )
    validator = staticmethod(ie_pps_checksum)
